File: src/MutualFund.cpp
#include "MutualFund.h"

MutualFund::MutualFund(const std::string& name, float value, float expenseRatio)
    : FinancialAsset(name, value), expenseRatio(expenseRatio) {}

float MutualFund::calculateAnnualReturn() const {
    return value * (1 - expenseRatio);
}

void MutualFund::displayAsset() const {
    std::cout << "Mutual Fund: " << name << ", Value: $" << value
              << ", Expense Ratio: " << expenseRatio << std::endl;
}

File: demo/src/main/java/com/example/MutualFund.java
package com.example;

public class MutualFund extends FinancialAsset {
    private float expenseRatio;

    public MutualFund(String name, float value, float expenseRatio) {
        super(name, value);
        this.expenseRatio = expenseRatio;
    }

    @Override
    public float calculateAnnualReturn() {
        return value * expenseRatio;
    }

    @Override
    public void displayAsset() {
        System.out.println("Mutual Fund: " + name + ", Value: $" + value + ", Expense Ratio: " + expenseRatio);
    }
}
